yellow-paper/docs/contracts/main.md

```diff
@@ -0,0 +1,319 @@
+---
+title: Cross-chain communication
+description: How the L1 contracts facilitate cross-chain communication
+---
+
+This section are to describe what our L1 contracts do, what each of them are responsible for and how they interact with the circuits. 
+
+Note that the only reason that we even have any contracts is to facilitate cross-chain communication. The contracts are not required for the rollup to function, but required to bridge assets and to reduce the cost of light nodes.
+
+:::info Assumption
+- Blocks and their content have been defined in earlier so we will not go into detail on that here.
+:::
+
+We have multiple contracts, some that exist already, and some that are to be build. The collection of these contracts is called the L1 contracts and it was makes up the rollup precense on L1.  
+
+**Existing contracts**:
+- Registry
+- State transitioner
+- Inbox
+- Outbox
+
+**To be build contracts**
+- Sequencer Selection contract(s)
+- Governance/Upgrade contract(s)
+
+## Purpose of contracts
+The purpose of the L1 contracts are simple:
+- Facilitate cross-chain communication such that L1 liquidity can be used on L2
+- Act as a validating light node for L2 that every L1 node implicitly run
+
+## Registry
+The registry is a contract that holds the addresses of the other contracts. It is a simple contract that keeps track of where the other contracts are, it is to give the nodes a single source of truth for where the other contracts are. Depending on the scheme might be used to handle upgrades, seen as the point of entry to the protocol from a node perspective - it is the only contract that the nodes MUST know about. It can tell them about the rest, and where they need to look for blocks etc. 
+
+Must keep track of current and historical addresses of the contracts. 
+
+## State transitioner
+The state transitioner is the contract that handles the state and perform the state transition on L1. It makes sure that the proof is valid and that L1 is updated accordingly. It is the main component of our fully validating light node.
+
+Whenever a block is received it is to:
+- Compute the "starting" state of the block (the state before the block is applied)
+- Check that the starting state is the same as the current state
+- Check the global variables provided (block number, timestamp, version, chainId)
+    - The block number MUST be the next block number
+    - The timestamp MUST:
+        - be newer than the previous block inclusion
+        - not be in the future
+    - The version MUST be the same as the current version
+    - The chainId MUST be the same as the current chainId
+- Compute the "ending" state of the block (the state after the block is applied)
+- Compute a single public inputs hash from the block content
+    - - **Circuit:** MUST match the computation of the circuit
+- Validate the proof
+- Update state root to the ending state
+- Consume the L1 to L2 messages specified in the block from the inbox
```

Comment:
since an Inbox can be for multiple versions do we add:
`Consume the L1 to L2 messages specified in the block from the inbox for the particular version`

Reply:
If the inbox is for multiple version, there is not an inbox for a particular version, its shared. There is just a specific recipient rollup.